owncloudApp/src/main/java/com/owncloud/android/workers/OldLogsCollectorWorker.kt

```diff
@@ -0,0 +1,86 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Abel García de Prada
+ * Copyright (C) 2021 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.workers
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.owncloud.android.MainApp
+import com.owncloud.android.data.storage.LocalStorageProvider
+import timber.log.Timber
+import java.io.File
+import java.io.IOException
+import java.util.Calendar
+import java.util.concurrent.TimeUnit
+
+class OldLogsCollectorWorker(
+    val appContext: Context,
+    workerParameters: WorkerParameters
+) : CoroutineWorker(
+    appContext,
+    workerParameters
+) {
+
+    override suspend fun doWork(): Result {
+        val logsDirectory = getLogsDirectory()
+        val logsFiles = getLogsFiles(logsDirectory)
+
+        return try {
+            removeOldLogs(logsFiles)
+            Result.success()
+        } catch (ioException: IOException) {
```

Comment:
If IOException and SecurityException are the only two exceptions you could combine both cases to a simple
``` kotlin
catch(exception: Exception) {
   Result.failure()
}
```

Reply:
I agree in part. 

Catching them separately lets us handle each error in a different way. And we know exactly which are the potential errors.

In this case, that we don't do anything specific for each error, we could unify them, but I would keep them separately anyway.